### PR TITLE
Homogeneize gradle and maven behavior around Formatter issues on a sp…

### PR DESCRIPTION
**Pull request by blacelle**
This would fix https://github.com/diffplug/spotless/issues/868 by reporting explicitly a faulty file in gradle-plugin. SIUch error reporting is already present in maven plugin.

### CODE REVIEW

File: plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java [hunk 1]
```diff
@@ -94,26 +94,34 @@ public void performAction(InputChanges inputs) throws Exception {
...
+			} catch (RuntimeException e) {
+				throw new IllegalArgumentException("Issue processing file: " + input, e);
+			}
 		}
 		if (dirtyState.isClean()) {
 			// Remove previous output if it exists
 			Files.deleteIfExists(output.toPath());
 		} else if (dirtyState.didNotConverge()) {
-			getLogger().warn("Skipping '" + input + "' because it does not converge.  Run {@code spotlessDiagnose} to understand why");
+			getLogger().warn("Skipping '{}}' because it does not converge.  Run {@code spotlessDiagnose} to understand why", input);
```
Comment (nedtwigg) [inline plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java:114]: Extra `}`

File: plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java [hunk 2]
```diff
@@ -94,26 +94,34 @@ public void performAction(InputChanges inputs) throws Exception {
...
 			dirtyState = PaddedCell.isClean();
 		} else {
-			dirtyState = PaddedCell.calculateDirtyState(formatter, input);
+			try {
+				dirtyState = PaddedCell.calculateDirtyState(formatter, input);
+			} catch (IOException e) {
+				throw new IOException("Issue processing file: " + input, e);
+			} catch (RuntimeException e) {
+				throw new IllegalArgumentException("Issue processing file: " + input, e);
+			}
```
Comment (nedtwigg) [inline plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java:108]: I'm fine with this, but I'm curious why it is needed. Are you parsing these? Do we need a `SpotlessException` with a field `File`?

**Review by nedtwigg — CHANGES_REQUESTED** [review]
Great, thanks! Minor changes requested below.

File: plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java (hunk 2 above)
Comment (blacelle) [reply plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java:108]: Nope x2. The point is just to get the fileName in the Exception message. Added a unitTest

Comment (nedtwigg) [reply plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java:110]: FYI the test failure is real (Windows `\` vs unix `/`). Here's the method we've been using to detect win

https://github.com/diffplug/spotless/blob/f65d366d57cb66286a137827c09c6504d51414e3/lib/src/main/java/com/diffplug/spotless/FileSignature.java#L112-L115

Comment (blacelle) [reply plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java:110]: Thanks. I updated the test

**Review by nedtwigg — APPROVED** [review]

**Comment by taofik69** [comment]
#1615
--
BarreraVIP Jumat, 10 Maret 2023, 02:45PM +08:00 dari Ned Twigg  ***@***.*** :

**Pull request merged**